test/jake/jakelib/migrate-test.js

```diff
@@ -543,6 +589,60 @@ test('migrate:down undos only the last run migration', (temp) => {
     });
 });
 
+test('migrate:down <name> undos only the defined run migration', (temp) => {
+  const migrationsPath = `${temp}/migrations`;
+  const migrationFile1 = '001_one.js';
+  const migrationFile2 = '002_two.js';
+  const migrationData = `
+      exports.up = () => Promise.resolve();
+      exports.down = () => Promise.resolve();
+    `;
+
+  fs.writeFileSync(`${migrationsPath}/${migrationFile1}`, migrationData);
+
+  fs.writeFileSync(`${migrationsPath}/${migrationFile2}`, migrationData);
+
+  return assertExec(
+    `node ${KNEX} migrate:latest \
+    --client=sqlite3 \
+    --connection=${temp}/db \
+    --migrations-directory=${migrationsPath}`,
+    'run_all_migrations'
+  ).then(() => {
+    return assertExec(
+      `node ${KNEX} migrate:down ${migrationFile1} \
+      --client=sqlite3 \
+      --connection=${temp}/db \
+      --migrations-directory=${temp}/migrations`,
+      'undo_migration_001'
+    ).then(({ stdout }) => {
+      assert.include(
+        stdout,
+        `Batch 1 rolled back the following migrations:\n${migrationFile1}`
+      );
+      assert.notInclude(stdout, migrationFile2);
+    });
+  });
+});
+
+test('migrate:down <name> throw an error', (temp) => {
+  const migrationsPath = `${temp}/migrations`;
+  const migrationFile1 = '001_one.js';
+
+  return assertExec(
+    `node ${KNEX} migrate:down ${migrationFile1} \
```

Comment:
assertExecError is a better way to test negative scenarios

Reply:
Thank you, you are right. Done.